### ISSUE TITLE
Can `DynMetadata` be made constructable?

### ISSUE THREAD

**Issue by Rua** [rust-lang/rust#126828]
Related tracking issue: https://github.com/rust-lang/rust/issues/81513

As of right now, the only way to get a `DynMetadata` is to call `ptr::metadata` on an existing pointer, which can create a chicken-and-egg problem. It could be useful to be able to get the metadata for a hypothetical trait object `Foo as Trait` as well, without first having to construct such a trait object. However, I don't know if this would be possible, and whether it would lead to soundness issues or have other undesired consequences.

**Comment by bjorn3** [comment]
I think `std::ptr::metadata(std::ptr::null::<Foo>() as *const dyn Trait)` would work.

**Comment by spunit262** [comment]
Yes [metadata(null())](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=38f0a659073c5c6c46148c9f8d1013fc) does work.

`feature(unsize)` is only need if you want to be generic over the trait, otherwise `T: Trait` is enough.
